lifecycle-operator/controllers/common/schedulinggateshandler_test.go

```diff
@@ -0,0 +1,365 @@
+						{
+							Name: "gate",
+						},
+					},
+				},
+			},
+			wantError: false,
+			annotations: map[string]string{
+				apicommon.SchedulingGateRemoved: "true",
+			},
+		},
+		{
+			name:    "scheduling gates removed - not empty annotations",
+			podName: "pod",
+			pod: &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "pod",
+					Namespace: "default",
+					Annotations: map[string]string{
+						"test": "test",
+					},
+				},
+				Spec: corev1.PodSpec{
+					SchedulingGates: []corev1.PodSchedulingGate{
+						{
+							Name: "gate",
+						},
+					},
+				},
+			},
+			wantError: false,
+			annotations: map[string]string{
+				apicommon.SchedulingGateRemoved: "true",
+				"test":                          "test",
+			},
+		},
+	}
+
+	err := klcv1alpha3.AddToScheme(scheme.Scheme)
+	require.Nil(t, err)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := fake.NewClientBuilder().WithObjects(tt.pod).WithStatusSubresource(tt.pod).Build()
+			err := removePodGates(context.TODO(), client, tt.podName, tt.pod.Namespace)
+			if tt.wantError != (err != nil) {
+				t.Errorf("want error: %t, got: %v", tt.wantError, err)
+			}
+			if !tt.wantError {
+				pod := &corev1.Pod{}
+				err := client.Get(context.TODO(), types.NamespacedName{Namespace: tt.pod.Namespace, Name: tt.podName}, pod)
+				require.Nil(t, err)
+				require.Equal(t, []corev1.PodSchedulingGate(nil), pod.Spec.SchedulingGates)
+				require.Equal(t, tt.annotations, pod.Annotations)
+			}
+		})
+	}
+}
+
+func Test_GetPodsOfOwner(t *testing.T) {
+	namespace := "default"
+	tests := []struct {
+		name   string
+		uid    types.UID
+		kind   string
+		pods   *corev1.PodList
+		result []string
+	}{
+		{
+			name:   "pod list empty",
+			pods:   &corev1.PodList{},
+			result: nil,
+		},
+		{
+			name: "pod list not matching kind or uid",
+			pods: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "pod1",
+							Namespace: "default",
+							OwnerReferences: []v1.OwnerReference{
+								{
+									Kind: "unknown",
+									UID:  types.UID("uid"),
+								},
+							},
+						},
+					},
+				},
+			},
+			kind:   "unknown2",
+			uid:    types.UID("uid2"),
+			result: nil,
+		},
+		{
+			name: "pod list matches one pod of list",
+			pods: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "pod1",
+							Namespace: "default",
+							OwnerReferences: []v1.OwnerReference{
+								{
+									Kind: "unknown",
+									UID:  types.UID("uid"),
+								},
+							},
+						},
+					},
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "pod2",
+							Namespace: "default",
+							OwnerReferences: []v1.OwnerReference{
+								{
+									Kind: "unknown2",
+									UID:  types.UID("uid2"),
+								},
+							},
+						},
+					},
+				},
+			},
+			kind:   "unknown",
+			uid:    types.UID("uid"),
+			result: []string{"pod1"},
+		},
```

Comment:
Using the `PrependReactor` function of the fake client we could also add a case where the List request returns an error - I'll leave that up to you wether you want to include that in the PR though, as the error handling in the tested function is fairly trivial anyway

Reply:
https://github.com/kubernetes/client-go/blob/34d1b64e67dbec9cbfa9628109bdd7c913897f49/testing/fake.go#L103